R/plssem.R

```diff
@@ -0,0 +1,1888 @@
+        "varname" = unlist(unique_per_group),
+        stringsAsFactors = FALSE
+      )
+
+      rhoCmmdf <- data.frame(
+        "grpname" = rep(names(rhoCmmli[[1]]), vapply(unique_per_group, length, 0)),
+        "varname" = unlist(unique_per_group),
+        stringsAsFactors = FALSE
+      )
+
+      rhoCWeightedmmdf <- data.frame(
+        "grpname" = rep(names(rhoCWeightedmmli[[1]]), vapply(unique_per_group, length, 0)),
+        "varname" = unlist(unique_per_group),
+        stringsAsFactors = FALSE
+      )
+
+
+      for (mod_idx in seq_along(rhoTli)) {
+        for (grpname in names(rhoTli[[1]])) {
+
+          grp_idx <- which(rhoTdf[["grpname"]] == grpname)
+
+          row_idx <- grp_idx[match(names(rhoTli[[mod_idx]][[grpname]]), rhoTdf[grp_idx, "varname"])]
+
+          rhoTdf[row_idx, mod_idx + 2] <- rhoTli[[mod_idx]][[grpname]]
+          rhoCmmdf[row_idx, mod_idx + 2] <- rhoCmmli[[mod_idx]][[grpname]]
+          rhoCWeightedmmdf[row_idx, mod_idx + 2] <- rhoCWeightedmmli[[mod_idx]][[grpname]]
+        }
+      }
+
+
+      tabrho[["grp"]] <- rhoTdf[["grpname"]]
+      tabrho[["var"]] <- rhoTdf[["varname"]]
+      for (i in seq_along(rhoTli)) tabrho[[paste0("rhoT_", i)]] <- rhoTdf[[i + 2]]
+      for (i in seq_along(rhoCmmli)) tabrho[[paste0("rhoCmm_", i)]] <- rhoCmmdf[[i + 2]]
+      for (i in seq_along(rhoCWeightedmmli)) tabrho[[paste0("rhoCWeightedmm_", i)]] <- rhoCWeightedmmdf[[i + 2]]
+    }
+  }
+}
+
+.plsSemCor <- function(modelContainer, options, ready) {
+  if (!(options[["observedIndicatorCorrelation"]] || options[["impliedIndicatorCorrelation"]] ||
+        options[["observedConstructCorrelation"]] || options[["impliedConstructCorrelation"]]) ||
+      !is.null(modelContainer[["cors"]])) return()
+
+  cors <- createJaspContainer(gettext("Correlation tables"))
+  cors$position <- 3
+  cors$dependOn(c("observedIndicatorCorrelation",
+                    "impliedIndicatorCorrelation",
+                    "observedConstructCorrelation",
+                    "impliedConstructCorrelation",
+                    "models"))
+
+  modelContainer[["cors"]] <- cors
+
+  if (length(options[["models"]]) < 2) {
+    .plsSemCorTables(modelContainer[["results"]][["object"]][[1]], NULL, cors, options, ready)
+  } else {
+
+    for (i in seq_along(options[["models"]])) {
+      fit <- modelContainer[["results"]][["object"]][[i]]
+      name <- options[["models"]][[i]][["name"]]
+      .plsSemCorTables(fit, name, cors, options, ready)
+    }
+  }
+}
+
+.plsSemCorTables <- function(fit, name, parentContainer, options, ready) {
+  if (is.null(name)) {
+    corcont <- parentContainer
+  } else {
+    corcont <- createJaspContainer(name, initCollapsed = TRUE)
+  }
+
+  if (options[["group"]] == "") {
+
+    # without groups, these are tables
+
+    if (options[["observedIndicatorCorrelation"]]) {
+      oictab <- createJaspTable("Observed indicator correlation matrix")
```

Comment:
```suggestion
      oictab <- createJaspTable(gettext("Observed indicator correlation matrix"))
```
the `gettext`s are missing from other tables in this function as well

Reply:
also optionally, you could use `%setOrRetrieve%` (see `` ?jaspBase::`%setOrRetrieve%` ``) for generating the tables to make the code a bit more succinct